### PR TITLE
Add custom controls support

### PR DESCRIPTION
**Pull request by Lube**
Adding a prop customControls to render custom react elements.

Updated demo.

API:
```javascript
<RichTextEditor
  value={value}
  onChange={this._onChange}
  className="react-rte-demo"
  placeholder="Tell a story"
  toolbarClassName="demo-toolbar"
  editorClassName="demo-editor"
  readOnly={this.state.readOnly}
  customControls={[
    (handleChange, getValue) => {
      let choices = new Map(
        [
          {value: '1', label: 1},
          {value: '2', label: 2},
          {value: '3', label: 3}
        ].map((choice) => [choice.value, {label: choice.label}])
      );
      return (
        <ButtonGroup key={1}>
          <Dropdown
            choices={choices}
            selectedKey={getValue('some-state')}
            onChange={(val) => { handleChange('some-state', val); alert(val) }}
          />
        </ButtonGroup>
    )},
    <ButtonGroup key={2}>
      <IconButton
        label="Remove Link"
        iconName="remove-link"
        focusOnClick={false}
        onClick={() => alert('You pressed a button')}
      />
    </ButtonGroup>
  ]}
/>
```

### CODE REVIEW

**Comment by Lube** [comment]
I pushed the rest of the commits I had in another repo, I removed lib and dist from gitignore and changed webpack since the original one wouldnt build, let me know if I should rollback this in order to merge the pr

**Comment by pintocarlos** [comment]
Why did you need to gitignore the dist and build folders?
As far as rolling back and the webpack changes, the owner of the repo should make that call.

Also, I noticed that the build is still failing due to a type error with the flow static checker. Please correct that as well so you have a green build. Thanks!

**Comment by bherila** [comment]
Hi @Lube can you fix the merge conflict and CI so that we can try to get this merged?

Thanks,
Ben

**Comment by bherila** [comment]
@Lube I fixed the merge conflict, please fix the CI errors and I can review again this weekend.

File: src/lib/EditorToolbar.js [hunk 1]
```diff
@@ -98,6 +100,18 @@ export default class EditorToolbar extends Component {
 //         {this._renderBlockTypeDropdown()}
 //         {this._renderUndoRedo()}
         {buttonsGroups}
+        {customControls && customControls.map((f) => {
+          switch (typeof f) {
+            case 'function':
+              return f(
+                (key, value) => this.setState({['customControl' + key]: value}),
+                (key) => this.state['customControl' + key],
```
Comment (sstur) [inline src/lib/EditorToolbar.js]: This is an interesting approach. I'd prefer if we don't use string concatenation, but instead make an object for customControls that can be clearly typed.
```js
(key, value) => this.setState(
  ({customControls}) => ({customControls: {...customControls, [key]: value}})
),
(key) => this.state.customControls[key]
```

Then we can type state like so:
```js
type State = {
  showLinkInput: boolean;
  showImageInput: boolean;
  customControls: {[key: string]: string}
};
```

**Pull request merged**